File: en/java-spring-demo/src/main/java/com/itangcent/springboot/demo/common/dto/Node.java
package com.itangcent.springboot.demo.common.dto;

/**
 * Tree Node Data Transfer Object
 * Used to represent hierarchical data structures
 */
public class Node {
    /**
     * Node ID
     * Unique identifier
     */
    private Long id;

    /**
     * Node Name
     * Text identifier for display
     */
    private String name;

    /**
     * Parent Node
     * Points to the parent node of the current node, forming a tree structure
     */
    private Node parent;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Node getParent() {
        return parent;
    }

    public void setParent(Node parent) {
        this.parent = parent;
    }
} 

File: en/kotlin-spring-demo/src/main/kotlin/com/itangcent/common/dto/Node.kt
package com.itangcent.common.dto

/**
 * 树节点数据传输对象
 * 用于表示具有层级结构的数据
 */
data class Node(
    /**
     * 节点ID
     * 唯一标识符
     */
    var id: Long? = null,

    /**
     * 节点名称
     * 用于显示的文本标识
     */
    var name: String? = null,

    /**
     * 父节点
     * 指向当前节点的父节点，形成树形结构
     */
    var parent: Node? = null
) 